### PR TITLE
[drift_dev] Update analyzer and deprecated methods

### PR DESCRIPTION
**Pull request by Carapacik**
Also updated the minimum version of packages so that packages have null safety support. (#2073)

### CODE REVIEW

**Comment by Carapacik** [comment]
@simolus3, should I upgrade the drift_dev version to 2.1.1?

File: drift_dev/pubspec.yaml [hunk 1]
```diff
@@ -6,51 +6,51 @@ homepage: https://drift.simonbinder.eu/
 issue_tracker: https://github.com/simolus3/drift/issues
 
 environment:
-  sdk: '>=2.14.0 <3.0.0'
+  sdk: '>=2.17.0 <3.0.0'
```
Comment (simolus3) [inline drift_dev/pubspec.yaml:9]: > should I upgrade the drift_dev version to 2.1.1?

Actually it should be `2.2.0-dev`, there have been some new features in the meantime. I will strip the `-dev` before releasing, but you can also leave the version as it is.

In general, I don't like raising the minimum dependency constraints if we don't need it. It's fine for the language version I guess, but do we really need a new feature that has been added in `charcode` version `1.3.1`? If we support the lower version, our pubspec should indicate that as well. We want to test against the latest versions we claim to support, which is why the CI runs `pub upgrade`. There's no need to _only_ support the latest version by raising the constraint in the pubspec.

So, can we keep the constraints of the other packages? Given that we don't seem to be affected by any breaking changes in `analyzer`, I suggest doing this:

```dart
  analyzer: '>=4.5.0 < 6.0.0'
  analyzer_plugin: '^0.11.0'
```

If the lower bound has been wrong before and needed to be upgraded - that's fine! But I doubt we need to raise it to the latest version for every single dependency.

**Comment by simolus3** [comment]
Also, don't worry about the breaking CI for now. I'll have to fix that dependency issue.

**Comment by Carapacik** [comment]
Ok, I can fix version and some packages

**Comment by Carapacik** [comment]
@simolus3, to resolve conflicts in #2073, you need an analyzer version higher than 4.7.0. And the analyzer from version 4.2.0 already supports the minimum Dart SDK 2.17.

**Comment by simolus3** [comment]
Now it seems that to resolve the lints I need to use `5.0.0` as a minimum version :D Oh well...

**Comment by Carapacik** [comment]
I think it is necessary to update all dependencies in all projects to Dart 2.17. Because analyzer 5 requires this.

**Comment by Carapacik** [comment]
Maybe you should also ask to update the analyzer in build_value_generator?

In the meantime, you can just write 
```yaml
dependency_overrides:
  analyzer: ^5.0.0
```

**Comment by Carapacik** [comment]
@simolus3, check last commit here

**Comment by simolus3** [comment]
> I think it is necessary to update all dependencies in all projects to Dart 2.17. Because analyzer 5 requires this.

I'm not sure I don't understand what you mean with this. If a package has `sdk: '>=2.12.0 <3.0.0'` in its pubspec, it can still be used with Dart 2.17. So yeah, if we depend on `analyzer: ^5.0.0`, the user will have to be on Dart 2.17 or later, but IMO there's no need to spell that out in the pubspec.

File: examples/with_built_value/pubspec.yaml [hunk 2]
```diff
@@ -3,13 +3,16 @@ publish_to: none
...
   built_value: ^8.0.0
 
 dev_dependencies:
   drift_dev: ^2.0.0-dev
-  built_value_generator: ^8.1.1
+  built_value_generator: ^8.1.3
   build_runner: ^2.0.0
+
+dependency_overrides:
+  analyzer: ^5.0.0
```
Comment (simolus3) [inline examples/with_built_value/pubspec.yaml:20]: Could you add a comment here explaining why the override is necessary? Something like

```suggestion
dependency_overrides:
  # `built_value_generator` doesn't support the latest analyzer yet
  analyzer: ^5.0.0
```

**Comment by Carapacik** [comment]
I added a description to override

**Review by simolus3 — APPROVED** [review]
LGTM now, thanks for your contribution!

**Pull request merged**